keras_cv/metrics/object_detection/mean_box_count_delta.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+import tensorflow.keras as keras
+import tensorflow.keras.initializers as initializers
+
+from keras_cv import bounding_box
+
+
+class MeanBoxCountDelta(keras.metrics.Metric):
+    """MeanBoxCountDelta counts the difference of counts of predicted and true boxes.
+
+    MeanBoxCountDelta looks at the number of boxes in the ground truth dataset
+    and counts the delta between that number and the number of boxes your object
+    detection model predicted for that image. This is primarily useful when
+    attempting to tune the confidence threshold of your
+    `MultiClassNonMaxSuppression` layer in an object detection model.  If this
```

Comment:
This feels weird -- perhaps something other than `Mean` is appropriate so that it's more obvious whether there are too many or too few boxes?

Reply:
Yeah... I've thought about this too.  Sum doesn't work, as if you predict [-5] too few boxes one image, then [5] too many the next you get `0` which looks correct.  

Any better ideas?  Maybe this just means the name is bad?

Reply:
I added a `relative` mode to try to fix this issue.  You can have both metrics now: a relative one, and an absolute one.  This will help you run down issues in all cases.